sdl_viewer/src/graphic.rs

```diff
@@ -60,9 +60,9 @@ impl GlBuffer {
         GlBuffer { id }
     }
 
-    pub fn bind(&self) {
+    pub fn bind(&self, buffer_type: GLuint) {
```

Comment:
typesafty: this should be a parameter of the constructor of this buffer. Or is ever possible that a buffer will switch its type? 

I see two design here: Either make two new() functions, i.e. `new_array_buffer` and `new_element_array_buffer` or pass an enum with two possible elements into the constructor. I think I like the first one slightly better. What do you think?